experimental/traffic-portal/src/app/api/cache-group.service.ts

```diff
@@ -162,6 +162,77 @@ export class CacheGroupService extends APIService {
 		return this.delete<ResponseDivision>(`divisions/${id}`).toPromise();
 	}
 
+	public async getRegions(): Promise<Array<ResponseRegion>>;
+	public async getRegions(nameOrID: string | number): Promise<ResponseRegion>;
+
+	/**
+	 * Gets an array of regions from Traffic Ops.
+	 *
+	 * @param nameOrID If given, returns only the Region with the given name
+	 * (string) or ID (number).
+	 * @returns An Array of Region objects - or a single Region object if 'nameOrID'
+	 * was given.
+	 */
+	public async getRegions(nameOrID?: string | number): Promise<Array<ResponseRegion> | ResponseRegion> {
+		const path = "regions";
+		if(nameOrID) {
+			let params;
+			switch (typeof nameOrID) {
+				case "string":
+					params = {name: nameOrID};
+					break;
+				case "number":
+					params = {id: String(nameOrID)};
```

Comment:
Hm. Looks like I incorrectly typed the params argument as `Record<string, string>`. It should be `@angular/common/http.HttpParams | Record<string, string | number | boolean | ReadonlyArray<string | number | boolean>>`. You can fix that if you want, but it could wind up being a very involved change given how many places use it, but if you'd rather leave the cast for now that can be fixed at a later date.

Reply:
I'll leave it for now then.